src/QuartoNotebookWorker/src/render.jl

```diff
@@ -26,9 +36,15 @@ function _render_thunk(
     thunk::Base.Callable,
     code::AbstractString,
     cell_options::AbstractDict = Dict{String,Any}(),
+    is_expand_ref::Ref{Bool} = Ref(false),
 )
     captured, display_results = with_inline_display(thunk, cell_options)
-    if get(cell_options, "expand", false) === true
+
+    is_expand = is_expandable(captured.value) || get(cell_options, "expand", false) === true
```

Comment:
True, and it's more convenient. Maybe a little more magic, though. Is this on a similar level of magic as those observable js functions?

Reply:
I'm alright with this level of magic.

Reply:
if we go that way, would it make sense to refactor the mechanism itself? If we're already extending methods from a package, we might think about something more convenient or explicit than the `NamedTuple` mechanism? We chose that because it needed no dependencies, but the extension mechanism would require a weak dep anyway, so we could also use a type.

Reply:
Agreed.